apps/ledger-live-mobile/src/components/WebPTXPlayer/index.tsx

```diff
@@ -153,14 +179,17 @@ export const WebPTXPlayer = ({ manifest, inputs, disableHeader }: Props) => {
           const manifestId = url.searchParams.get("goToManifest");
 
           if (manifestId && goToURL) {
-            const flowName = url.searchParams.get("flowName") || "buy";
+            const flowName = url.searchParams.get("flowName");
 
             await AsyncStorage.setItem("manifest-id", manifestId);
-            await AsyncStorage.setItem("flow-name", flowName);
-            await AsyncStorage.setItem("last-screen", url.searchParams.get("lastScreen") || "");
-
-            navigation.navigate(NavigatorName.Exchange, {
-              screen: flowName === "buy" ? ScreenName.ExchangeBuy : ScreenName.ExchangeSell,
+            await AsyncStorage.setItem("flow-name", flowName || "buy");
+            await AsyncStorage.setItem(
+              "last-screen",
+              url.searchParams.get("lastScreen") || flowName || "",
+            );
```

Comment:
this becomes confusing

Reply:
Mentioned this morning, that the "Buy/Sell" page should display different messages depending on the origin (e.g., shortcut or quotes). However, for the "Card" page, the content is always the same regardless of the origin.

To address this, if last-screen is not set, it will be assigned a value from flow-name. This approach aims to avoid the need to manually specify last-screen or flow-name each time we use last-screen.